### PR TITLE
refactor: use next_back and match

### PR DESCRIPTION
**Pull request by GreasySlug**

### CODE REVIEW

**Comment by mtshiba** [comment]
This implementation is incorrect. Because `.iter().next_back()` always returns the same element. Also, `VecDeque`'s reverse iterator uses `next_back()` internally, so there is no need to change the iteration part.

File: crates/els/file_cache.rs [hunk 1]
```diff
@@ -82,9 +82,12 @@ impl FileCache {
     ) -> ELSResult<Option<Token>> {
         let Some(index) = self.get_token_index(uri, pos)? else {
             let tokens = self.get(uri)?.token_stream.as_ref().ok_or("lex error")?;
-            for token in tokens.iter().rev() {
-                if !token.is(TokenKind::EOF) && !token.is(TokenKind::Dedent) && !token.is(TokenKind::Newline) {
-                    return Ok(Some(token.clone()));
+            while let Some(token) = tokens.iter().next_back(){
```
Comment (GreasySlug) [inline crates/els/file_cache.rs:85]: > so there is no need to change the iteration part

Is this change also wrong?
```rust
            while let Some(token) = tokens.pop_back() {
```

```rust
    fn pop_back(&mut self) -> Option<T> {
        self.ref_mut_payload().pop_back()
    }
```

**Review by mtshiba — COMMENTED** [review]
Since `token_stream` is reusable, its ownership should not be taken away. Also, borrowing only is sufficient for code purposes.

**Comment by mtshiba** [comment]
In general, element traversal is most efficient with a for statement. This is because it allows optimization for boundary checks. This is also true when using VecDeque.

**Comment by GreasySlug** [comment]
Okay, this PR was meant to be a discussion, so, I'll close it

**Pull request closed without merging**